hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/PersistentStateStore.java

```diff
@@ -57,7 +57,11 @@ public PersistentStateStore(HdfsFrameworkConfig hdfsFrameworkConfig) {
   public void setFrameworkId(Protos.FrameworkID id) {
 
     try {
-      hdfsStore.setRawValueForId(FRAMEWORK_ID_KEY, id.toByteArray());
+      if (id != null) {
```

Comment:
Instead of setting the raw value for the ID to null, wouldn't it be preferable to remove it from the ZK State store instead?